client/my-sites/checkout/composite-checkout/test/checkout-contact-step.js

```diff
@@ -269,12 +265,13 @@ describe( 'Checkout contact step', () => {
 		render( <MyCheckout cartChanges={ cartChanges } />, container );
 		// Wait for the cart to load
 		await screen.findByText( 'Country' );
-		// Wait for the validation to complete
-		await waitForElementToBeRemoved( () => screen.queryAllByText( 'Please wait…' ) );
-		expect( screen.queryByTestId( 'payment-method-step--visible' ) ).toBeInTheDocument();
+		// Wait for the validation to complete, then check we've advanced to the next payment step
+		await waitFor( () =>
+			expect( screen.queryByTestId( 'payment-method-step--visible' ) ).toBeInTheDocument()
+		);
```

Comment:
I'm pretty sure we can change this to `findBy...` and exclude the `waitFor`. The [docs explain](https://testing-library.com/docs/queries/about/#types-of-queries) that this is what `findBy...` does. Combining it with `expect` is a little less clear to me, but I think it's this:

```suggestion
		expect( await screen.findByTestId( 'payment-method-step--visible' ) ).toBeInTheDocument();
```